src/app/components/shared/header/header.component.ts

```diff
@@ -1,19 +1,62 @@
 import { Component, OnInit } from '@angular/core';
 import { MenuItem } from 'primeng/api';
 import { OverlayService } from '../../../service';
-import { SingInComponent } from '../../sing-in/sing-in.component';
+import { SignInComponent } from '../../sign-in/sign-in.component';
+import { SignInService } from '../../../service';
+import { PopUpComponent } from '../../popup/popup.component';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
 })
+
 export class HeaderComponent implements OnInit {
   items: MenuItem[] = [];
+  isLoggedIn: boolean = false;
+
+  signInMenuItem: MenuItem = {
```

Comment:
To avoid the mess before constructor, it would be better to define properties here. And values can be set in some method (for isntance, initMenuItems) and this method can be called from the constructor (this.items = [ also can be moved there)

Reply:
Fixed